simulator/main.rs

```diff
@@ -152,74 +538,94 @@ fn main() {
     env.io.print_stats();
 }
 
-fn process_connection(env: &mut SimulatorEnv, conn: &mut Rc<Connection>) -> Result<()> {
-    let management = env.rng.gen_ratio(1, 100);
-    if management {
-        // for now create table only
-        maybe_add_table(env, conn)?;
-    } else if env.tables.is_empty() {
-        maybe_add_table(env, conn)?;
-    } else {
-        let roll = env.rng.gen_range(0..100);
-        if roll < env.opts.read_percent {
-            // read
-            do_select(env, conn)?;
-        } else if roll < env.opts.read_percent + env.opts.write_percent {
-            // write
-            do_write(env, conn)?;
+fn property_insert_select(env: &mut SimulatorEnv, conn: &mut Rc<Connection>) {
+    // Get a random table
+    let table = env.rng.gen_range(0..env.tables.len());
+
+    // let table = &env.tables[table];
+
+    // Pick a random column
+    let column_index = env.rng.gen_range(0..env.tables[table].columns.len());
+    let column = &env.tables[table].columns[column_index].clone();
```

Comment:
```suggestion
    let table = &env.tables[table];

    // Pick a random column
    let column_index = env.rng.gen_range(0..table.columns.len());
    let column = &table.columns[column_index].clone();
```

Reply:
This change cases a problem with the borrow checker, at least as far as I was able to work with it.